kuiper-game/Repositories/JsonFileSolarSystemRepository.cs

```diff
@@ -10,63 +11,72 @@ namespace Kuiper.Repositories
 {
     public class JsonFileSolarSystemRepository : ISolarSystemRepository
     {
-        private List<CelestialBody> solarSystem;
-        private FileInfo jsonFile;
-        public IEnumerable<CelestialBody> GetSolarSystem()
-        {
-            if (solarSystem.Count == 0) {
-                solarSystem = GetAllBodies(jsonFile);
-            }
-            return solarSystem;
-        }
+        private readonly string _filePath;
 
-        public JsonFileSolarSystemRepository(FileInfo jsonFile)
+        private static readonly Dictionary<string, Type> _typeMap = new()
         {
-            if (!jsonFile.Exists) throw new FileNotFoundException($"Could not find {jsonFile.FullName}");
-            this.jsonFile = jsonFile;
+            {"Planet", typeof(Planet)},
+            {"Star", typeof(Star)},
+            {"DwarfPlanet", typeof(DwarfPlanet)},
+            {"GasGiant", typeof(GasGiant)},
+            {"Moon", typeof(Moon)},
+            {"Asteroid", typeof(Asteroid)},
```

Comment:
Consider something like 
```suggestion
            {nameof(Asteroid), typeof(Asteroid)},
```